From: http://gcc.gnu.org/bugzilla/show_bug.cgi?id=15718

===================================================================
RCS file: /cvs/gcc/gcc/gcc/config/alpha/alpha.h,v
retrieving revision 1.185.4.5
retrieving revision 1.185.4.6
diff -u -r1.185.4.5 -r1.185.4.6
--- gcc/gcc/config/alpha/alpha.h	2004/05/28 00:02:03	1.185.4.5
+++ gcc/gcc/config/alpha/alpha.h	2004/08/27 00:01:15	1.185.4.6
@@ -647,6 +647,7 @@
 #define HARD_REGNO_MODE_OK(REGNO, MODE) 				\
   ((REGNO) >= 32 && (REGNO) <= 62 					\
    ? (MODE) == SFmode || (MODE) == DFmode || (MODE) == DImode		\
+     || (MODE) == SCmode || (MODE) == DCmode				\
    : 1)
 
 /* Value is 1 if MODE is a supported vector mode.  */
